pyproject.toml

```diff
@@ -1,78 +1,85 @@
+    "msgspec>=0.19.0,<0.20",
+    "numpy<2.1",
+    "protobuf>=5.29.3,<6",
+    "zstandard>=0.23.0,<0.24",
+    "click>=8.1.8,<9",
+    "httpx>=0.28.1,<0.29",
+]
 
-[tool.poetry.dependencies]
-python = '>=3.10,<3.13'
-pytz = '^2023.3.post1'
-trio = '^0.22.2'
-Pillow = '^10.0.1'
-docker = '^6.1.3'
-py-leap = {git = 'https://github.com/guilledk/py-leap.git', rev = 'v0.1a32'}
-toml = '^0.10.2'
-msgspec = "^0.19.0"
-numpy = "<2.1"
-protobuf = "^5.29.3"
-zstandard = "^0.23.0"
-click = "^8.1.8"
-httpx = "^0.28.1"
+[project.scripts]
+skynet = "skynet.cli:skynet"
+txt2img = "skynet.cli:txt2img"
+img2img = "skynet.cli:img2img"
+upscale = "skynet.cli:upscale"
+inpaint = "skynet.cli:inpaint"
 
-[tool.poetry.group.frontend]
-optional = true
+[dependency-groups]
+frontend = [
+    "triopg>=0.6.0,<0.7",
+    "aiohttp>=3.8.5,<4",
+    "psycopg2-binary>=2.9.7,<3",
+    "pyTelegramBotAPI>=4.14.0,<5",
+    "discord.py>=2.3.2,<3",
+]
+dev = [
+    "pdbpp>=0.10.3,<0.11",
+    "pytest>=7.4.2,<8",
+    "pytest-trio>=0.8.0,<0.9",
+]
+cuda = [
+    "torch==2.5.1+cu121",
+    "scipy==1.15.1",
+    "numba==0.60.0",
+    "quart>=0.19.3,<0.20",
+    "triton==3.1.0",
+    "xformers>=0.0.29,<0.0.30",
+    "hypercorn>=0.14.4,<0.15",
+    "diffusers==0.32.1",
+    "quart-trio>=0.11.0,<0.12",
+    "torchvision==0.20.1+cu121",
+    "accelerate==0.34.0",
+    "transformers==4.48.0",
+    "huggingface-hub>=0.27.1,<0.28",
+    "invisible-watermark>=0.2.0,<0.3",
+    "bitsandbytes>=0.45.0,<0.46",
+    "basicsr>=1.4.2,<2",
+    "realesrgan>=0.3.0,<0.4",
+    "sentencepiece>=0.2.0",
+]
 
-[tool.poetry.group.frontend.dependencies]
-triopg = {version = '^0.6.0'}
-aiohttp = {version = '^3.8.5'}
-psycopg2-binary = {version = '^2.9.7'}
-pyTelegramBotAPI = {version = '^4.14.0'}
-'discord.py' = {version = '^2.3.2'}
+[tool.uv]
+default-groups = [
+    "frontend",
+    "dev",
+    "cuda",
+]
 
-[tool.poetry.group.dev]
-optional = true
+[[tool.uv.index]]
+name = "torch"
+url = "https://download.pytorch.org/whl/cu121"
+explicit = true
 
-[tool.poetry.group.dev.dependencies]
-pdbpp = {version = '^0.10.3'}
-pytest = {version = '^7.4.2'}
-pytest-trio = "^0.8.0"
-
-[tool.poetry.group.cuda]
-optional = true
-
-[tool.poetry.group.cuda.dependencies]
-torch = {version = '2.5.1+cu121', source = 'torch'}
-scipy = {version = '1.15.1'}
-numba = {version = '0.60.0'}
-quart = {version = '^0.19.3'}
-triton = {version = '3.1.0', source = 'torch'}
-xformers = {version = '^0.0.29'}
-hypercorn = {version = '^0.14.4'}
-diffusers = {version = '0.32.1'}
-quart-trio = {version = '^0.11.0'}
-torchvision = {version = '0.20.1+cu121', source = 'torch'}
-accelerate = {version = '0.34.0'}
-transformers = {version = '4.48.0'}
-huggingface-hub = {version = '^0.27.1'}
-invisible-watermark = {version = '^0.2.0'}
-bitsandbytes = "^0.45.0"
-basicsr = "^1.4.2"
-realesrgan = "^0.3.0"
-
-[[tool.poetry.source]]
-name = 'torch'
-url = 'https://download.pytorch.org/whl/cu121'
-priority = 'explicit'
+[tool.uv.sources]
```

Comment:
Yeah only thing i might tweak (and i realize this wasn't your decision since you used the converter but) maybe reordering the sources entries versus index entries so that you can kinda read the de-referencing top to bottom? 

Obvi not a big deal :)